### PR TITLE
Add unit tests for com.util.CodeUtil

### PR DESCRIPTION
**Pull request by EricHetti**
I've analysed your codebase and noticed that `com.util.CodeUtil` is not fully tested.
I've written some tests for the methods in this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests will help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other classes that you consider important.

### CODE REVIEW

File: src/tests/java/com/util/CodeUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    @Test
+    public void testDbEncode() {
+        Assert.assertNull(CodeUtil.dbEncode(null));
+        Assert.assertEquals("foo", CodeUtil.dbEncode("foo"));
+    }
+
+    @Test
+    public void testDbDecode() {
+        Assert.assertNull(CodeUtil.dbDecode(null));
+        Assert.assertEquals("foo", CodeUtil.dbDecode("foo"));
```
Comment (freddytuxworth) [inline src/tests/java/com/util/CodeUtilTest.java]: should have some tests for `dbEncode` and `dbDecode` which show characters being encoded/decoded

**Review by freddytuxworth — CHANGES_REQUESTED** [review]

File: src/tests/java/com/util/CodeUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        char[] in = {'t', '\\', '\\', 'p', '\\', 'u', '0', 'a', 'A', 'a'};
+        char[] convtBuf = {'\u0139', '\u02f9', '\u01e1', '\u001e', 'p', '@', 'U'};
+
+        Assert.assertEquals("\\pપ",
+                codeUtil.fromUnicode(in, 1, 8, convtBuf));
+    }
+
+    @Test
+    public void testFromUnicode2() {
+        CodeUtil codeUtil = new CodeUtil();
```
Comment (JohnLBergqvist) [inline src/tests/java/com/util/CodeUtilTest.java]: I don't think there's a need to use an entirely new object & test here. You could have them as part of the 1st test.

**Review by JohnLBergqvist — CHANGES_REQUESTED** [review]

File: src/tests/java/com/util/CodeUtilTest.java (hunk 1 above)
Comment (Braavos6) [reply src/tests/java/com/util/CodeUtilTest.java]: Done

File: src/tests/java/com/util/CodeUtilTest.java (hunk 2 above)
Comment (Braavos6) [reply src/tests/java/com/util/CodeUtilTest.java]: Done

File: src/tests/java/com/util/CodeUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    @Test
+    public void testToHtml() {
+        Assert.assertEquals("", CodeUtil.toHtml(null));
+        Assert.assertEquals("foo", CodeUtil.toHtml("foo"));
+    }
+
+    @Test
+    public void testToUnicode() {
+        Assert.assertEquals("1",
```
Comment (petersear) [inline src/tests/java/com/util/CodeUtilTest.java]: I think lines 45 +46 can be joined and be < 80 chars

File: src/tests/java/com/util/CodeUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                CodeUtil.toUnicode("1", true));
+        Assert.assertEquals(">\\ \\\\\\u0000\\r\\t\\n\\f\\!",
+                CodeUtil.toUnicode("> \\\u0000\r\t\n\f!", true));
+    }
+
+    @Test
+    public void testFromUnicode() {
+        CodeUtil codeUtil = new CodeUtil();
+        char[] in1 = {'t', '\\', '\\', 'p', '\\', 'u', '0', 'a', 'A', 'a'};
+        char[] convtBuf1 = {'\u0139', '\u02f9', '\u01e1', '\u001e', 'p', '@', 'U'};
```
Comment (petersear) [inline src/tests/java/com/util/CodeUtilTest.java]: line > 80 chars

**Review by petersear — CHANGES_REQUESTED** [review]

File: src/tests/java/com/util/CodeUtilTest.java (hunk 3 above)
Comment (Braavos6) [reply src/tests/java/com/util/CodeUtilTest.java]: Done

File: src/tests/java/com/util/CodeUtilTest.java (hunk 4 above)
Comment (Braavos6) [reply src/tests/java/com/util/CodeUtilTest.java]: Line 55 is 75 chars

Comment (petersear) [reply src/tests/java/com/util/CodeUtilTest.java]: plus 8 spaces :-)

Comment (Braavos6) [reply src/tests/java/com/util/CodeUtilTest.java]: Apologies.. file has been changed

File: src/tests/java/com/util/CodeUtilTest.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+    @Test
+    public void testFromUnicode() {
+        CodeUtil codeUtil = new CodeUtil();
+        char[] in1 = {'t', '\\', '\\', 'p', '\\', 'u', '0', 'a', 'A', 'a'};
+        char[] convtBuf1 = {'\u0139', '\u02f9',
+                '\u01e1', '\u001e', 'p', '@', 'U'};
+
+        char[] in2 = {'\\', '0', '\\', 'r', '\\', 't', '\\', 'f', '\\', 'n'};
+        char[] convtBuf2 = {'\u0139', '\u02f9', '\u0120',
```
Comment (JohnLBergqvist) [inline src/tests/java/com/util/CodeUtilTest.java]: The 4 `char[]`s should be `final`.

And i'd declare the 2nd set after the asserts from the 1st set.

**Review by JohnLBergqvist — CHANGES_REQUESTED** [review]

File: src/tests/java/com/util/CodeUtilTest.java (hunk 5 above)
Comment (Braavos6) [reply src/tests/java/com/util/CodeUtilTest.java]: Done and done. I separated them to make it look nice :)

**Review by JohnLBergqvist — APPROVED** [review]

**Review by freddytuxworth — APPROVED** [review]

File: src/tests/java/com/util/CodeUtilTest.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    public void testToUnicode() {
+        Assert.assertEquals("1", CodeUtil.toUnicode("1", true));
+        Assert.assertEquals(">\\ \\\\\\u0000\\r\\t\\n\\f\\!",
+                CodeUtil.toUnicode("> \\\u0000\r\t\n\f!", true));
+    }
+
+    @Test
+    public void testFromUnicode() {
+        CodeUtil codeUtil = new CodeUtil();
+        final char[] in1 = {'t', '\\', '\\', 'p', '\\', 'u', '0', 'a', 'A', 'a'};
```
Comment (petersear) [inline src/tests/java/com/util/CodeUtilTest.java]: line > 80 chars (apologies, there are two that I mentioned verbally but didn't flag here. Caused by the addition of `final`)

File: src/tests/java/com/util/CodeUtilTest.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        final char[] convtBuf1 = {'\u0139', '\u02f9',
+                '\u01e1', '\u001e', 'p', '@', 'U'};
+
+        Assert.assertEquals("\\pપ",
+                codeUtil.fromUnicode(in1, 1, 8, convtBuf1));
+
+        final char[] in2 = {'\\', '0', '\\', 'r', '\\', 't', '\\', 'f', '\\', 'n'};
```
Comment (petersear) [inline src/tests/java/com/util/CodeUtilTest.java]: line > 80 chars

**Review by petersear — CHANGES_REQUESTED** [review]

File: src/tests/java/com/util/CodeUtilTest.java (hunk 6 above)
Comment (Braavos6) [reply src/tests/java/com/util/CodeUtilTest.java]: Forgot to push changes.. Thanks for the reminder

File: src/tests/java/com/util/CodeUtilTest.java (hunk 7 above)
Comment (Braavos6) [reply src/tests/java/com/util/CodeUtilTest.java]: Forgot to push changes.. Thanks for the reminder

**Review by petersear — APPROVED** [review]

**Pull request closed without merging**